x-pack/plugins/security_solution/common/endpoint/data_generators/endpoint_action_generator.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DeepPartial } from 'utility-types';
+import { merge } from 'lodash';
+import { BaseDataGenerator } from './base_data_generator';
+import { EndpointActionData, ISOLATION_ACTIONS } from '../types';
+
+interface EcsError {
+  code: string;
+  id: string;
+  message: string;
+  stack_trace: string;
+  type: string;
```

Comment:
@pzl
These follow the ECS `Error` fields, not sure if `code`, `id` and `type` have a union of specific string types?